### PR TITLE
Add skip doc ms for each artifact. 

### PR DESCRIPTION
**Pull request by sima-zhu**
Remove artifact.options to flatten artifact structure.
Added feature of enable/disable DocMs step per artifact level.
Issue:https://github.com/Azure/azure-sdk-tools/issues/922

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-net-release.yml [hunk 1]
```diff
@@ -235,7 +235,7 @@ stages:
                           # we override the regular script path because we have cloned the build tools repo as a separate artifact.
                           ScriptPath: 'eng/common/scripts/copy-docs-to-blobstorage.ps1'
 
-          - ${{if ne(artifact.options.skipUpdatePackageVersion, 'true')}}:
+          - ${{if ne(artifact.skipUpdatePackageVersion, 'true')}}:
             - deployment: UpdatePackageVersion
               displayName: "Update Package Version"
               condition: and(succeeded(), ne(variables['Skip.UpdatePackageVersion'], 'true'))
```
Comment (scbedd) [inline eng/pipelines/templates/stages/archetype-net-release.yml:241]: In other PRs, we swapped the optional `Integration` publish to the dev feed under `skipPublishDevFeed`. 

image

That screenshot is from JS. I kinda figure we should update everything if we're doing a triggering update.

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-net-release.yml:241]: I search all nested structure in the release yml. That's all we need to update. Do you suggest anything else?

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-net-release.yml:241]: .NET is not having artifact level publish dev feed. So ignore in .net.

Comment (weshaggard) [reply eng/pipelines/templates/stages/archetype-net-release.yml:241]: Yeah we can ignore it for the dev publish for now until a time were we might need it.

**Review by weshaggard — APPROVED** [review]

**Review by scbedd — APPROVED** [review]

**Pull request merged**